sharding/shard.go

```diff
@@ -0,0 +1,208 @@
+package sharding
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Shard base struct.
+type Shard struct {
+	shardDB *shardKV
```

Comment:
Can you change this to a common interface? Right now, it only accepts a database of struct shardKV, so it’s not very swappable.